app/controllers/users/passkeys_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Users
+  class PasskeysController < ApplicationController
+    before_action :ensure_feature_enabled!
+
+    def index
+      @create_passkey_options = WebAuthn::Credential.options_for_create(
+        user: {
+          id: current_user.webauthn_id,
+          name: current_user.email
+        },
+        exclude: current_user.passkeys.pluck(:external_id),
+        authenticator_selection: {
+          resident_key: 'required',
+          user_verification: 'required'
+        }
+      )
+      session[:current_registration_challenge] = { challenge: @create_passkey_options.challenge }
+    end
+
+    def create
+      webauthn_passkey = WebAuthn::Credential.from_create(JSON.parse(params[:passkey_public_key]))
+
+      begin
+        webauthn_passkey.verify(session[:current_registration_challenge]["challenge"], user_verification: true)
```

Comment:
nit:

Si guardas la session asi:
```ruby
session[:creation_challenge] = @create_passkey_options.challenge
```

Despues podes hacer
```suggestion
        webauthn_passkey.verify(session[:creation_challenge], user_verification: true)
```